File: python/palindromic_substrings.py
class Solution(object):
    def countSubstrings(self, s):
        """
        :type s: str
        :rtype: int
        """
        N = len(s)
        result = 0

        for center in range(2 * N - 1):
            left = center / 2
            right = left + center % 2

            while (left >= 0 and
                   right < N and
                   s[left] == s[right]):
                left -= 1
                right += 1
                result += 1
        return result

File: javascript/palindromic-substrings.js
/**
 * @param {string} s
 * @return {number}
 */
let countSubstrings = function(s) {
    let ans = 0;
    let extendSearch = (s, left, right) => {
        while (left >= 0 &&
        right < s.length &&
        s.charAt(left) === s.charAt(right)) {
            left--;
            right++;
            ans++;
        }
    }

    for (let i = 0, len = s.length; i < len; i++) {
        extendSearch(s, i, i);
        extendSearch(s, i, i + 1);
    }

    return ans;
};